tests/kernel/userbufer_validate/src/userbuffer_validate.c

```diff
@@ -0,0 +1,352 @@
+	x86_mmu_set_pte_rw_bit((void *)0x00406000, 1);
+	status = _x86_mmu_buffer_validate((void *)0x00404000,
+			0x3000, 0x1); /* should pass */
+	if (status != 0) {
+		TC_PRINT("multi_page_writeable failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+static int multi_page_user_supervisor_rw(void)
+{
+	/* user/supervisor test */
+	x86_mmu_set_pte_us_bit((void *)0x00405000, 0); /* supervisor */
+	x86_mmu_set_pte_us_bit((void *)0x00406000, 1); /* user */
+	x86_mmu_set_pte_us_bit((void *)0x00404000, 0); /* supervisor */
+	status = _x86_mmu_buffer_validate((void *)0x00405000,
+			0x3000, 0x1 | 0x2); /* should fail */
+	if (status != -EPERM) {
+		TC_PRINT("multi_page_user_supervisor_rw failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+static int multi_page_supervisor_rw(void)
+{
+	x86_mmu_set_pte_us_bit((void *)0x00404000, 0); /* supervisor */
+	x86_mmu_set_pte_us_bit((void *)0x00405000, 0); /* supervisor */
+	x86_mmu_set_pte_us_bit((void *)0x00406000, 0); /* supervisor */
+	status = _x86_mmu_buffer_validate((void *)0x00404000,
+			0x3000, 0x1 | 0x0); /* should pass */
+	if (status != 0) {
+		TC_PRINT("multi_page_supervisor_rw failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+static int multi_page_user_rw(void)
+{
+	x86_mmu_set_pde_us_bit((void *)0x00405000, 1); /* set pde user */
+	x86_mmu_set_pte_us_bit((void *)0x00404000, 1); /* user */
+	x86_mmu_set_pte_us_bit((void *)0x00405000, 1); /* user */
+	x86_mmu_set_pte_us_bit((void *)0x00406000, 1); /* user */
+	status = _x86_mmu_buffer_validate((void *)0x00404000,
+			0x3000, 0x1 | 0x2); /* should pass */
+	if (status != 0) {
+		TC_PRINT("multi_page_user_rw failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+static int multi_page_supervisor_rw_user(void)
+{
+	x86_mmu_set_pde_us_bit((void *)0x00405000, 1); /* pde set to user */
+	x86_mmu_set_pte_us_bit((void *)0x00404000, 0); /* supervisor */
+	x86_mmu_set_pte_us_bit((void *)0x00405000, 0); /* supervisor */
+	x86_mmu_set_pte_us_bit((void *)0x00406000, 0); /* supervisor */
+	status = _x86_mmu_buffer_validate((void *)0x00404000,
+			0x3000, 0x1 | 0x2); /* should fail */
+	if (status != -EPERM) {
+		TC_PRINT("multi_page_supervisor_rw_user failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+static int multi_page_supervisor_rw_supervisor(void)
+{
+	x86_mmu_set_pde_us_bit((void *)0x00405000, 0); /* set pde supervisor */
+	x86_mmu_set_pte_us_bit((void *)0x00404000, 0); /* supervisor */
+	x86_mmu_set_pte_us_bit((void *)0x00405000, 0); /* supervisor */
+	x86_mmu_set_pte_us_bit((void *)0x00406000, 0); /* supervisor */
+	status = _x86_mmu_buffer_validate((void *)0x00404000,
+			0x3000, 0x1 | 0x2); /* should fail */
+	if (status != -EPERM) {
+		TC_PRINT("multi_page_supervisor_rw_supervisor failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+void test_buffer_rw_read(void)
+{
+	zassert_true(buffer_rw_read() == TC_PASS, NULL);
+}
+void test_buffer_writeable_write(void)
```

Comment:
always break a line before creating a new function. Apply that everywhere

Reply:
1 line break applied for all functions.